plugins/inputs/azure_monitor/azure_monitor.go

```diff
@@ -59,8 +63,20 @@ func (am *AzureMonitor) SampleConfig() string {
 
 // Init is for setup, and validating config.
 func (am *AzureMonitor) Init() error {
+	var clientOptions = azcore.ClientOptions{Cloud: cloud.AzurePublic}
+	if len(am.CloudOption) != 0 {
+		switch cloudOption := am.CloudOption; cloudOption {
+		case "AzureChina":
+			clientOptions = azcore.ClientOptions{Cloud: cloud.AzureChina}
+		case "AzureGovernment":
+			clientOptions = azcore.ClientOptions{Cloud: cloud.AzureGovernment}
+		case "AzurePublic":
+			clientOptions = azcore.ClientOptions{Cloud: cloud.AzurePublic}
+		}
+	}
```

Comment:
Instead of checking the length, we can just check for the empty case and set it to Azure Public.

Also there should be a default case for when the value was not recognized.

Reply:
I've adjusted the code base.